File: backend/app/schemas/claim.py
# backend/app/schemas/claim.py

from datetime import datetime

from pydantic import BaseModel, Field

from app.models.enums import (
    AttendanceStatus,
    ClaimStatus,
)


class ClaimCreateRequest(BaseModel):
    attendance_id: int
    reason: str = Field(
        min_length=1,
        max_length=2000,
    )


class ClaimResolutionRequest(BaseModel):
    resolution_reason: str = Field(
        min_length=1,
        max_length=2000,
    )


class ClaimResponse(BaseModel):
    id: int
    attendance_id: int
    student_id: int
    original_attendance_status: AttendanceStatus
    status: ClaimStatus
    reason: str
    created_at: datetime

    class Config:
        from_attributes = True


class ClaimDetailResponse(BaseModel):
    id: int
    attendance_id: int
    student_id: int
    original_attendance_status: AttendanceStatus
    reason: str
    status: ClaimStatus

    claim_resolved_by: int | None = None
    claim_resolved_at: datetime | None = None
    claim_resolution_reason: str | None = None

    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ClaimStatisticsResponse(BaseModel):
    total: int
    pending: int
    approved: int
    rejected: int

File: mobile_app/lib/features/claims/domain/claim.dart
// mobile_app/lib/features/claims/domain/claim.dart

import 'claim_status.dart';

class Claim {
  final int id;

  final int attendanceId;

  final int studentId;

  final String reason;

  final ClaimStatus status;

  final DateTime createdAt;

  final String? resolutionReason;

  final DateTime? resolvedAt;

  const Claim({
    required this.id,
    required this.attendanceId,
    required this.studentId,
    required this.reason,
    required this.status,
    required this.createdAt,
    this.resolutionReason,
    this.resolvedAt,
  });

  factory Claim.fromJson(
    Map<String, dynamic> json,
  ) {
    return Claim(
      id: json['id'] as int,
      attendanceId: json['attendance_id'] as int,
      studentId: json['student_id'] as int,
      reason: json['reason'] as String,
      status: claimStatusFromApi(
        json['status'] as String,
      ),
      createdAt: DateTime.parse(
        json['created_at'] as String,
      ),
      resolutionReason:
          json['claim_resolution_reason'] as String?,
      resolvedAt:
          json['claim_resolved_at'] != null
              ? DateTime.parse(
                  json['claim_resolved_at']
                      as String,
                )
              : null,
    );
  }
}